forge-game/src/main/java/forge/game/cost/CostPayment.java

```diff
@@ -136,7 +131,12 @@ public final void refundPayment() {
 
     public boolean payCost(final CostDecisionMakerBase decisionMaker) {
         adjustedCost = CostAdjustment.adjust(cost, ability);
-        final List<CostPart> costParts = adjustedCost.getCostPartsWithZeroMana();
+        List<CostPart> costParts = adjustedCost.getCostPartsWithZeroMana();
+
+        if (adjustedCost.getCostParts().size() > 1 && decisionMaker.getPlayer().getController().isFullControl(FullControlFlag.ChooseCostOrder)) {
+            // if mana part is shown here it wouldn't include reductions, but that's just a minor inconvenience
+            costParts = decisionMaker.getPlayer().getController().orderCosts(costParts);
+        }
```

Comment:
I would make the `.isFullControl(FullControlFlag.ChooseCostOrder)` check inside the `orderCosts` function for this controller